src/main/kotlin/com/gaarnik/blacksmithtycoon/menu/AbstractMenu.kt

```diff
@@ -21,13 +21,13 @@ abstract class AbstractMenu(owner: InventoryHolder?, size: Int, title: String) {
     }
 
     fun onInventoryClick(e: InventoryClickEvent) {
-        val close = onMenuItemClick(e.rawSlot)
+        val close = onMenuItemClick(e.whoClicked as Player, e.rawSlot)
```

Comment:
ce cast est pas safe, faudrait faire un check au début de la méthode pour être secure
`if (e.whoClicked !is Player) return`
Du coup tu vas avoir un smart cast sur ta ligne si je dis pas de bêtises